app/cdap/components/ConnectionList/index.tsx

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, styled, Typography } from '@material-ui/core';
+import { exploreConnection } from 'components/Connections/Browser/GenericBrowser/apiHelpers';
+import { getCategorizedConnections } from 'components/Connections/Browser/SidePanel/apiHelpers';
+import { fetchConnectors } from 'components/Connections/Create/reducer';
+import { GCSIcon } from 'components/ConnectionList/iconStore';
+import * as React from 'react';
+import { useState } from 'react';
+import { useLocation, useParams } from 'react-router';
+import SubHeader from './Components/SubHeader';
+import ConnectionsTabs from './Components/ConnectionTabs';
+import { useStyles } from './styles';
+import If from 'components/shared/If';
+import LoadingSVG from 'components/shared/LoadingSVG';
+import ErrorSnackbar from 'components/SnackbarComponent';
+
+const SelectDatasetWrapper = styled(Box)({
+  overflowX: 'scroll',
+  display: 'flex',
+  borderTop: '1px solid #E0E0E0;',
+
+  height: '100%',
+  '& > :first-child': {
+    minWidth: '280px',
+  },
+  '& > :not(:first-child)': {
+    minWidth: '300px',
+  },
+});
+
+const DatasetWrapper = () => {
+  const { connectorType } = useParams() as any;
+
+  const classes = useStyles();
+  const loc = useLocation();
+  const queryParams = new URLSearchParams(loc.search);
+  const pathFromUrl = queryParams.get('path') || '/';
+  const [loading, setLoading] = useState(true);
+  const [isErrorOnNoWorkspace, setIsErrorOnNoWorkSpace] = useState<boolean>(false);
+
+  const toggleLoader = (value: boolean, isError?: boolean) => {
+    setLoading(value);
+    if (isError) {
+      setIsErrorOnNoWorkSpace(isError);
+    }
+  };
+  let connectionId;
+  const [dataForTabs, setDataForTabs] = useState([
+    {
+      data: [],
+      showTabs: true,
+      selectedTab: null,
+      isSearching: false,
+    },
+  ]);
+
+  const getConnectionsTabData = async () => {
+    let connectorTypes = await fetchConnectors();
```

Comment:
can you add some comments explaining this block of code

Reply:
Done